### PR TITLE
add flathub-beta

### LINKED ISSUE #34: please make use of flathub-beta channel for test builds
Could we please make an unofficial test build from master, to test recent Qt6 port?
For that, [flathub-beta](https://discourse.flathub.org/t/how-to-use-flathub-beta/2111) repository exists.
Could we make use of it?

> The way this works is that each GitHub repository builds the master branch for the stable repository, which will have the flatpak branch name “stable”, and then the beta git branch will build into the beta repository with the flatpak branch “beta”.

https://blogs.gnome.org/alexl/2019/02/19/changes-in-flathub-land/

So essentially we need to create "beta" branch here which points to master or some commit there.

@Ferenc-

### PR DESCRIPTION
**Pull request by bam80**
Fixes #34 

@Ferenc- @biodranik please create a "beta" branch here so I could re-target my PR there

### CODE REVIEW

**Comment by biodranik** [comment]
Does it work?

**Comment by bam80** [comment]
Yes. Could you please make a "beta" branch and I'll make a PR for it?

**Comment by biodranik** [comment]
Done. How are you planning to use the beta branch?

**Comment by bam80** [comment]
The idea is to make an announce on the main side, with a "call to testing".

Then testers do the following:
```
flatpak remote-add --user --if-not-exists flathub-beta https://flathub.org/beta-repo/flathub-beta.flatpakrepo
flatpak install --user https://flathub.org/beta-repo/appstream/app.organicmaps.desktop.flatpakref
flatpak run --user  app.organicmaps.desktop//beta
```
It will install the beta app alongside with stable and run it.
After that, the app can be updated from the beta channel as usual.
 
Only one app will be shown in the menu, so to run beta one would need to do it from CLI or switch the repos:
https://discourse.flathub.org/t/how-to-use-flathub-beta/2111

**Comment by bam80** [comment]
> Done. How are you planning to use the beta branch?

The beta branch is needed here in this repo, not in the main one) Sorry I didn't make it clear.

File: changelog-2023.06.20beta-1.patch [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
++++ b/packaging/app.organicmaps.desktop.metainfo.xml
+@@ -113,6 +113,14 @@
+   </screenshots>
+ 
+   <releases>
++   <release version="2023.06.20-1 beta" date="2023-06-20">
++     <description>
++       <p>Highlights:</p>
++         <ul>
++           <li>Ported to Qt6</li>
```
Comment (bam80) [inline changelog-2023.06.20beta-1.patch:13]: Please tell if we need more info for the beta.

**Comment by Ferenc-** [comment]
@bam80 Awesome, that you took the initiative on this.

I think, I have already found the first regression:
* As opposed to the `stable` flatpak branch this `test` branch now broke compatibility with PP OG.
The `OpenGLES3` api gets selected, when the device is clearly only capable of `ES 2` and not `ES 3`, and consequently the shader code which is meant for `ES 3` fails to compile.

I am not sure yet if this is a consequence of https://github.com/organicmaps/organicmaps/pull/2466, or another commit,
still need bisecting.

**Comment by bam80** [comment]
@Ferenc- Thanks for the feedback!
Do you think we could mention the found problem for mobile as known and still release the beta so others could test it on desktop meanwhile? It will be on separate channel anyway..

File: changelog-2023.06.20beta-1.patch (hunk 1 above)
Comment (Ferenc-) [reply changelog-2023.06.20beta-1.patch:13]: IMHO no, and we can add it later if we change our minds.

**Comment by Ferenc-** [comment]
> @Ferenc- Thanks for the feedback! Do you think we could mention the found problem for mobile as known and still release the beta so others could test it on desktop meanwhile? It will be on separate channel anyway..

OK, I have opened https://github.com/flathub/app.organicmaps.desktop/issues/36 for tracking it and pinned it on  https://github.com/flathub/app.organicmaps.desktop/issues so testers hopefully see it before flooding us with duplicate issues.

**Comment by bam80** [comment]
Thanks.
BTW, are you aware of `flatpak-bisect`?

**Comment by bam80** [comment]
We should have squashed the commits..

**Comment by bam80** [comment]
For some reason, the old version occurred in the Beta:
```
  $ flatpak run  app.organicmaps.desktop//beta
LOG TID(1) INFO   1.2979e-05 qt/main.cpp:123 main(): Organic Maps 2023.06.04-1-acc7c054-dirty-Linux started, detected CPU cores: 4
```
Any ideas?

**Comment by bam80** [comment]
I opened an issue upstream:
https://github.com/flathub/flathub/issues/4268

File: app.organicmaps.desktop.yml [hunk 2]
```diff
@@ -23,10 +23,10 @@ modules:
     sources:
       - type: git
         url: https://github.com/organicmaps/organicmaps
-        tag: 2023.06.04-13-android
-        commit: acc7c0547db4285dd8841ae7f98811268e38d908
+        branch: master
+        commit: be5a0aa2991377eedd61f19ec3853332b51b114b
```
Comment (hfiguiere) [inline app.organicmaps.desktop.yml:27]: this will break as soon as there is a new commit.

Comment (bam80) [reply app.organicmaps.desktop.yml:27]: #37

**Pull request merged**